### PR TITLE
soundfont-arachno: init at 1.0

### PR DESCRIPTION
**Pull request by mrtnvgr**
## Description of changes

This PR adds Arachno Soundfont to nixpkgs

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: lib/licenses.nix [hunk 1]
```diff
@@ -118,6 +118,11 @@ in mkLicense lset) ({
     fullName = "Apache License 2.0 with LLVM Exceptions";
   };
 
+  arachnosoft = {
+    fullName = "Arachnosoft License";
+    url = "http://www.arachnosoft.com/main/download.php?id=soundfont-sf2"; # license included in sources
+  };
```
Comment (alyssais) [inline lib/licenses.nix:124]: I think people (and programs) will expect the license URL to be a place where the text of the license can be seen, not a software download, so if this license is not online at all, it probably makes more sense to just use `licenses.free` or `licenses.unfree`.

**Review by h7x4 — CHANGES_REQUESTED** [review]
Just a few nits.

Also, please rebase your commits so that the commit log looks somewhat like this:

```
maintainers: add mrtnvgr
soundfont-arachno: init at 1.0
```

**Review by h7x4 — APPROVED** [review]
- [X] package path fits guidelines
- [X] package name fits guidelines
- [X] package version fits guidelines
- [X] package build on x86_64-linux
- [X] executables tested on x86_64-linux
- [X] `meta.description` is set and fits guidelines
- [X] `meta.license` fits upstream license
- [X] `meta.platforms` is set
- [X] `meta.maintainers` is set
- [X] build time only dependencies are declared in `nativeBuildInputs`
- [X] source is fetched using the appropriate function
- [X] the list of `phases` is not overridden
- [X] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
- [X] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [X] patches that are remotely available are fetched rather than vendored

Result of `nixpkgs-review pr 258963` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>soundfont-arachno</li>
  </ul>
</details>

Built and tested with fluidsynth and https://en.wikipedia.org/wiki/File:MIDI_sample.mid, seems to work fine.

File: pkgs/data/soundfonts/arachno/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{ lib, stdenvNoCC, fetchzip }:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "arachno";
+  version = "1.0";
+
+  src = fetchzip {
+    # Linked on http://www.arachnosoft.com/main/download.php?id=soundfont-sf2:
+    url = "https://www.dropbox.com/s/2rnpya9ecb9m4jh/arachno-soundfont-${builtins.replaceStrings ["."] [""] version}-sf2.zip";
+    sha256 = "sha256-Z5ETe0AKPCi4KlM2xOlNcyQn1xvCuor3S/tcrF+AwNQ=";
```
Comment (eclairevoyant) [inline pkgs/data/soundfonts/arachno/default.nix:10]:
```suggestion
    hash = "sha256-Z5ETe0AKPCi4KlM2xOlNcyQn1xvCuor3S/tcrF+AwNQ=";
```

File: pkgs/data/soundfonts/arachno/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  };
+
+  installPhase = ''
+    runHook preInstall
+    install -Dm444 Arachno*.sf2 $out/share/soundfonts/arachno.sf2
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "General MIDI-compliant bank, aimed at enhancing the realism of your MIDI files and arrangements.";
```
Comment (eclairevoyant) [inline pkgs/data/soundfonts/arachno/default.nix:21]:
```suggestion
    description = "General MIDI-compliant bank, aimed at enhancing the realism of your MIDI files and arrangements";
```

<https://nixos.org/manual/nixpkgs/unstable/#var-meta-description>

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -30178,6 +30178,8 @@ with pkgs;
 
   sorts-mill-goudy = callPackage ../data/fonts/sorts-mill-goudy { };
 
+  soundfont-arachno = callPackage ../data/soundfonts/arachno { };
```
Comment (eclairevoyant) [inline pkgs/top-level/all-packages.nix:30181]: would be better to put this in `pkgs/by-name`, and we can remove this line

**Review by eclairevoyant — CHANGES_REQUESTED** [review]
Thanks for the contribution! A few changes below:

**Review by eclairevoyant — APPROVED** [review]

**Comment by eclairevoyant** [comment]
Oh should probably update the PR and commit name, since you updated the package name

**Comment by mrtnvgr** [comment]
> Oh should probably update the PR and commit name, since you updated the package name

What name would be better `soundfont-arachno` or `arachno`?

**Comment by eclairevoyant** [comment]
`soundfont-arachno` seems more searchable, and other packages in nixpkgs use that naming style

**Pull request merged**